routing.go

```diff
@@ -317,20 +317,17 @@ func (dht *IpfsDHT) getValues(ctx context.Context, key string, stopQuery chan st
 				})
 
 				rec, peers, err := dht.protoMessenger.GetValue(ctx, p, key)
-				switch err {
-				case routing.ErrNotFound:
-					// in this case, they responded with nothing,
-					// still send a notification so listeners can know the
-					// request has completed 'successfully'
-					routing.PublishQueryEvent(ctx, &routing.QueryEvent{
-						Type: routing.PeerResponse,
-						ID:   p,
-					})
-					return nil, err
-				default:
+				if err != nil && err != internal.ErrInvalidRecord {
```

Comment:
Hm. I liked this way of doing thing. Which tool has a problem with it?

Reply:
staticcheck didn't like that `default` wasn't the first or the last. I can restore it to a `switch` if you prefer.

Reply:
Eh, up to you. Merge when ready.

Reply:
Switched it back to a `switch`.